packages/dialect-react-sdk/hooks/useAddresses.ts

```diff
@@ -0,0 +1,398 @@
+import useDialectSdk from './useDialectSdk';
+import useDialectWallet from './useDialectWallet';
+
+interface CreateUpdateParams {
+  type: AddressType;
+  value: string;
+}
+
+interface DeleteParams {
+  type: AddressType;
+}
+
+interface ToggleParams {
+  type: AddressType;
+  enabled: boolean;
+}
+
+interface VerifyParams {
+  type: AddressType;
+  code: string;
+}
+interface ResendParams {
+  type: AddressType;
+}
+
+type AddressActionParams =
+  | CreateUpdateParams
+  | DeleteParams
+  | ToggleParams
+  | VerifyParams
+  | ResendParams;
+
+type AddressEnriched = Address & {
+  enabled: boolean;
+  dappAddress?: DappAddress;
+};
+interface UseAddressesValue {
+  addresses: Record<AddressType, AddressEnriched> | Record<string, never>;
+  create: (params: CreateUpdateParams) => Promise<void>;
+  update: (params: CreateUpdateParams) => Promise<void>;
+  delete: (params: DeleteParams) => Promise<void>;
+  toggle: (params: ToggleParams) => Promise<void>;
+
+  verify: (params: VerifyParams) => Promise<void>;
+  resend: (params: ResendParams) => Promise<void>;
+
+  isCreatingAddress: boolean;
+  isUpdatingAddress: boolean;
+  isDeletingAddress: boolean;
+  isSendingCode: boolean;
+  isVerifyingCode: boolean;
+
+  isFetching: boolean;
+  errorFetching: DialectSdkError | null;
+}
+
+interface UseAddressesParams {
+  refreshInterval?: number;
+}
+
+function useAddresses({
+  refreshInterval,
+}: UseAddressesParams = EMPTY_OBJ): UseAddressesValue {
+  const { wallet: walletsApi } = useDialectSdk();
+  const { dappAddress: dappPublicKey } = useDialectDapp();
+  const { connected: isWalletConnected } = useDialectWallet();
+  const [isCreatingAddress, setCreatingAddress] = useState(false);
+  const [isUpdatingAddress, setUpdatingAddress] = useState(false);
+  const [isDeletingAddress, setDeletingAddress] = useState(false);
+  const [isVerifyingCode, setVerifyingCode] = useState(false);
+  const [isSendingCode, setSendingCode] = useState(false);
+
+  // Fetch all wallet addresses (it doesn't include a enabled prop)
+  const {
+    data: addresses,
+    error: errorFetchingAddresses = null,
+    mutate: mutateAddresses,
+  } = useSWR(
+    WALLET_ADDRESSES_CACHE_KEY_FN(walletsApi),
+    walletsApi ? () => walletsApi.addresses.findAll() : null,
+    {
+      refreshInterval,
+      refreshWhenOffline: true,
+    }
+  );
+
+  const isFetchingAddresses =
+    Boolean(walletsApi) && !errorFetchingAddresses && addresses === undefined;
+
+  const {
+    data: dappAddresses,
+    error: errorFetchingDappAddresses = null,
+    mutate: mutateDappAddresses,
+  } = useSWR(
+    WALLET_DAPP_ADDRESSES_CACHE_KEY_FN(walletsApi, dappPublicKey),
+    walletsApi && dappPublicKey
+      ? () =>
+          walletsApi.dappAddresses.findAll({
+            addressIds: addresses?.map((addr) => addr.id),
+          })
+      : null,
+    {
+      refreshInterval,
+      refreshWhenOffline: true,
+    }
+  );
+
+  const isFetchingDappAddresses =
```

Comment:
can you explain why is this like this?

In my mind this doesn't reflect what really happens
I understand the case you want to handle on the ui side, but in my opinion this should reflect the real process behind api